### PR TITLE
Add EIP: SLOT precompile

### PR DESCRIPTION
**Pull request by Marchhill**
This EIP proposes to add a new precompile that returns the corresponding slot for the current block.

### CODE REVIEW

**Comment by daniellehrner** [comment]
Is there a reason why it needs to be precompile and not just a new op code?

**Comment by Marchhill** [comment]
> Is there a reason why it needs to be precompile and not just a new op code?

I have a bit on this in the rationale, it prevents the EVM from growing and allows L2s to decide whether they need to implement it. See also [this comment](https://ethereum-magicians.org/t/eip-7793-txindex-precompile/21513/4?u=marchhill)

File: EIPS/eip-slot-precompile.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+eip:
```
Comment (abcoathup) [inline EIPS/eip-slot-precompile.md:2]:
```suggestion
eip: 7843
```
Assigning next sequential EIP/ERC/RIP number.
Numbers are assigned by editors & associates.

Please also update the filename.

File: EIPS/eip-slot-precompile.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+title: SLOT precompile
+description: Precompile to get the current slot
+author: Marc Harvey-Hill (@Marchhill)
+discussions-to:
```
Comment (abcoathup) [inline EIPS/eip-slot-precompile.md:6]: Please create a discussions topic in Eth Magicians using the template: https://ethereum-magicians.org/c/eips/5

**Comment by g11tech** [comment]
seems like you have reverted everything with your latest commit

File: EIPS/eip-7843.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+It is currently possible to calculate the slot number from the block timestamp. However, this requires hardcoding the chain slot length into a smart contract. This would require the contract code to be changed in the event of a future change to slot length. A better approach is for the slot number to be calculated by the execution layer and exposed in a precompile.
+
+### Encrypted Mempools
+
+This precompile aims to improve support for encrypted mempools. In order to be secure, the validity of encrypted mempool transactions should be tied to the inclusion of all transactions by a proposer in the correct slot. This rule can be enforced by a smart contract using this precompile.
+
+## Specification
+
+If `block.number >= TBD` a new precompiled contract `SLOT` shall be created at address `TBD`.
```
Comment (g11tech) [inline EIPS/eip-7843.md:27]: the hardforks are now timestamp based

**Review by g11tech — APPROVED** [review]

**Pull request merged**